plugins/inputs/logfile/fileconfig.go

```diff
@@ -203,3 +251,55 @@ func (config *FileConfig) isMultilineStart(logValue string) bool {
 	}
 	return config.MultiLineStartPatternP.MatchString(logValue)
 }
+
+func (config *FileConfig) shouldFilterLog(event logs.LogEvent) bool {
+	if config.Filters == nil || len(config.Filters) < 1 {
+		return false
+	}
+	config.sampleCount += 1
+	start := time.Now()
+	for _, filter := range config.Filters {
+		if filter.ExpressionP == nil {
+			filter.ExpressionP = regexp.MustCompile(filter.Expression)
```

Comment:
`LogFilter.init()` already does `regexp.Compile()`. So why is it necessary to do `regexp.MustCompile()` here?
`MustCompile()` could `panic` - https://pkg.go.dev/regexp#MustCompile

Why not `continue` or `return false`?

Reply:
I _think_ I had this in here from testing where the `ExpressionP` was uninitialized. The reason I chose `MustCompile` here was because the expression should have been validated in the `init()` function, so there shouldn't be any risk of it panicking here. I will have to look over this again to see if I can take this out, since, like you said, the init function should already compile the expression.